packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+
+		if (!supportedRecordsNumber) {
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message:
+					"Door Lock Logging supported records number query timed out, skipping interview...",
+				level: "warn",
+			});
+			return;
+		}
+
+		const supportedNumberLogMessage = `received response for supported records number: ${supportedRecordsNumber}`;
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: supportedNumberLogMessage,
+			direction: "inbound",
+		});
+
+		setRecordMetadata.call(this, supportedRecordsNumber);
+
+		// Just retrieve the lastest record
+		const record = await api.getRecord(LATEST_RECORD_NUMBER_KEY);
+
+		if (!record) {
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message:
+					"Door Lock Logging record query timed out, skipping interview...",
+				level: "warn",
+			});
+			return;
+		}
+
+		const recordLogMessage = `received response for latest record: ${JSON.stringify(
+			record,
+		)}`;
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: recordLogMessage,
+			direction: "inbound",
+		});
+	}
+}
+
+@CCCommand(DoorLockLoggingCommand.RecordsSupportedReport)
+export class DoorLockLoggingCCRecordsSupportedReport extends DoorLockLoggingCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 1);
+
+		this.supportedRecordsNumber = this.payload[0];
+		this.persistValues();
+	}
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Max logging records stored",
+	})
+	public readonly supportedRecordsNumber: number;
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported records number": this.supportedRecordsNumber,
+			},
+		};
+	}
+
+	public persistValues(): boolean {
+		if (!super.persistValues()) return false;
+
+		this.getValueDB().setValue(
+			getDoorLockLoggingRecordsSupportedValueID(this.endpointIndex),
+			this.supportedRecordsNumber,
+		);
+
+		return true;
+	}
+}
+
+const convertEventTypeToLabel = (eventType: EventType): string => {
+	return eventTypeLabel[EventType[eventType] as keyof typeof EventType];
+};
+
+@CCCommand(DoorLockLoggingCommand.RecordsSupportedGet)
+@expectedCCResponse(DoorLockLoggingCCRecordsSupportedReport)
+export class DoorLockLoggingCCRecordsSupportedGet extends DoorLockLoggingCC {}
+
+@CCCommand(DoorLockLoggingCommand.RecordReport)
+export class DoorLockLoggingCCRecordReport extends DoorLockLoggingCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 11);
+
+		this.recordNumber = this.payload[0];
+		this.recordStatus = this.payload[5] >>> 5;
+		if (this.recordStatus === RecordStatus.Empty) {
```

Comment:
This condition (`recordStatus === 0xfe`) will never be true if you shift the value.

Reply:
It's checking against `0x00`, which should work.
I've changed the `HoldsLegalData` payload to reflect your comment.